### DISCUSSION TITLE
Get the HTML Template used in the Document Writer as an actual HTML

### DISCUSSION THREAD

**Discussion by NoraTheExplorer** [nextgenhealthcare/connect#4983] [Q&A]
Hi,

I am using the DocumentWriter to create a HTML file and I directly wrote the HTML code in the "HTML Template" field:
![image](https://user-images.githubusercontent.com/88822553/150980579-304bbcdc-15d0-4339-9120-6a071e16ad27.png)

So the creation of the PDF works flawless in 3.X.
But I thought about having the HTML I process at the end as, well, an HTML.

Is there a way to access the sent data? The template is not in the encoded data or transformed data at all.

I browsed through the discussions here and the old forum, also the [java docs](http://javadocs.mirthcorp.com/connect/3.12.0/user-api/com/mirth/connect/userutil/ImmutableConnectorMessage.html) but I couldn't find anything (except depracted getSent() functions).

**Comment by jonbartels** [comment]
1. For questions like this *always* look at the channel+connector maps in the message view. Each connector can populate its own intermediate objects to those map. Not all do, but it's always worth looking.
2. The relevant code in MC is `com.mirth.connect.connectors.doc.DocumentDispatcher#writeDocument` I don't see where it would be populating the template then storing the template content to a map or attachment.

I can't quite think of how but you may have to do something like build your HTML template in a separate destination then pass it as the template content to your document writer.
